src/main/java/winstone/Launcher.java

```diff
@@ -229,8 +230,12 @@ private void writePortToFileIfNeeded() throws IOException {
                     Path portFile = Paths.get(portFileName);
                     Path portDir = portFile.getParent();
                     Files.createDirectories(portDir);
-                    try (BufferedWriter writer = Files.newBufferedWriter(portFile, StandardCharsets.UTF_8)) {
+                    Path tmpPath = Files.createTempFile(portDir, "temp", "");
+                    try (BufferedWriter writer = Files.newBufferedWriter(tmpPath, StandardCharsets.UTF_8)) {
                         writer.write(Integer.toString(port));
+                        Files.move(tmpPath, portFile, StandardCopyOption.ATOMIC_MOVE);
```

Comment:
should we handle the `AtomicMoveNotSupportedException` thrown here?

(may have been hidden due to the existing `IOException` already thrown)

Reply:
Error handling has been added in [a60732f](https://github.com/jenkinsci/winstone/pull/249/commits/a60732f8dc91a82a418e9773d3a123389f9d318c), please review and suggest any improvements if you'll have one.